react-components/src/components/RuleBasedOutputs/utils.ts

```diff
@@ -469,17 +727,23 @@ const applyNodeStyles = (
 
   const nodeIndexSet = ruleOutputAndStyleIndex.styleIndex.getIndexSet();
   nodeIndexSet.clear();
-  treeNodes?.forEach((node) => {
+
+  const assetIds: number[] = [];
+
+  for (const node of treeNodes) {
     const range = new NumericRange(node.treeIndex, node.subtreeSize);
     nodeIndexSet.addRange(range);
-  });
```

Comment:
Is there a reason this was changed from `treeNodes?.forEach` to `for (... of ...)`? I usually prefer the first (although here we could even use `reduce`)

Reply:
The idea is to improve the performance at this phase. `For of`, in general, is faster than `forEach `depending on the browser engine. And it is way faster than `reduce` mostly. Actually I would prefer using` for loops` intead 😆, which is a bit faster than `for of`. When I changed these parts to regular `for/ofs`, I saw some improvements when generating the styling before sending to Reveal by comparing both cases in the flame graph.

Reply:
I see :thinking: I didn't expect there to be a big performance gap, but I suppose we should think about that as well